gammapy/modeling/models/spatial.py

```diff
@@ -775,6 +775,28 @@ def to_region(self, **kwargs):
             **kwargs,
         )
 
+    @classmethod
+    def from_region(cls, region, **kwargs):
+        """Create a `DiskSpatialModel from a ~regions.EllipseSkyRegion`
+
+        Parameters
+        ----------
+        region : `~regions.EllipseSkyRegion`
+            region to create model from
+
+        Returns
+        -------
+        spatial_model : `~gammapy.modeling.models.DiskSpatialModel`
+        """
+        frame = kwargs.pop("frame", region.center.frame)
+        model = cls.from_position(getattr(region.center, frame))
```

Comment:
Shouldn't `.from_position(position=region.center)` work directly? The frame handling is done in `.from_position()`...

Reply:
`.from_position()` takes the frame of the `SkyCoord` as the frame of the model. The regions can be defined in a different frame than that of the analysis, so added this...

Reply:
Oh, but this should be handled in Gammpay correctly (I hope)? It should be possible to define a model in ICRS while the maps are in galactic coordinates...so just copying over the frame from the region should be fine. If not we should make it work...

Reply:
Yes, we are not fully consistent here. However in this case the input is very well defined and only one class is possible. So I think the check is simple and helpful in this case...